Refactor headnav bar icons

wxiaoguang opened the pull request.

wxiaoguang commented:
![image](https://github.com/user-attachments/assets/d1ff918c-2523-4559-bd69-e4e7c1a44ffa)

wxiaoguang commented:
> Would swap position with the "+" icon.

That's very difficult by current layout. Feel free to propose changes and edit this PR directly.

silverwind commented:
I see, we have this complexity with the mobile layout.

silverwind commented:
We should probably remove this button in the user menu now?

<img width="198" alt="image" src="https://github.com/user-attachments/assets/a3337f88-cb10-472d-8e20-b0b87a4c2fad" />

silverwind commented:
Found a bug, opening `/notifications` gives this 500:

```
Render failed, failed to render template: user/notification/notification, error: template error: builtin(static):user/notification/notification_div:3:32 : executing "user/notification/notification_div" at <call .NotificationUnreadCount ctx>: error calling call: call of nil
----------------------------------------------------------------------
		{{$notificationUnreadCount := call .NotificationUnreadCount ctx}}

wxiaoguang commented:
> We should probably remove this button in the user menu now, given that it's duplicate?

Removed in https://github.com/go-gitea/gitea/pull/34922/commits/127274defc24a2ce6b550744ffcbe249f0b01e0e



> Found a bug, opening `/notifications` gives this 500:

Hmm, the page was covered by tests in `links_test.go`, but the tests were wrong and didn't really test. Fixed together in  https://github.com/go-gitea/gitea/pull/34922/commits/127274defc24a2ce6b550744ffcbe249f0b01e0e

wxiaoguang commented:
> so I'd prefer if we change this title to the same translation string.

I am not sure which one is better. Feel free to pick one to replace the other.
